File: leetcode/977.squares-of-a-sorted-array/go/sortedSquares_test.go
package sortedsquares

import (
	"fmt"
	"reflect"
	"testing"
)

type testcase struct {
	nums []int
	want []int
}

func TestSortedSquares(t *testing.T) {
	tcs := []testcase{
		{nums: []int{-4, -1, 0, 3, 10}, want: []int{0, 1, 9, 16, 100}},
		{nums: []int{-7, -3, 2, 3, 11}, want: []int{4, 9, 9, 49, 121}},
	}

	for _, tc := range tcs {
		t.Run(fmt.Sprintf("%v -> %v", tc.nums, tc.want), func(t *testing.T) {
			got := sortedSquares(tc.nums)
			if !reflect.DeepEqual(got, tc.want) {
				t.Fatalf("sortedsquare, expected: %v, got: %v", tc.want, got)
			}
		})
	}
}

File: leetcode/977.squares-of-a-sorted-array/ts/sortedsquares.test.ts
import { expect, test } from "vitest";
import sortedSquares from "./sortedsquares.js";

interface testcase {
    nums: number[]
    want: number[]
}

test("sortedsquare", () => {
    const tcs: testcase[] = [
        { nums: [- 4, -1, 0, 3, 10], want: [0, 1, 9, 16, 100] },
        { nums: [- 7, -3, 2, 3, 11], want: [4, 9, 9, 49, 121] },
        { nums: [- 3], want: [9] },
        { nums: [], want: [] },
    ]

    for (const tc of tcs) {
        expect(sortedSquares(tc.nums)).toEqual(tc.want)
    }
})
